src/main/java/billgates/usecases/insert_entry/InsertEntryController.java

```diff
@@ -0,0 +1,15 @@
+package billgates.usecases.insert_entry;
+
+import billgates.database.QueryEntryData;
+
+public class InsertEntryController {
+    private final InsertEntryInputPort useCase;
+
+    public InsertEntryController(InsertEntryInputPort useCase) {
+        this.useCase = useCase;
+    }
+
+    public void insert(int billId, QueryEntryData entry) {
```

Comment:
QueryEntryData should not be used here.

QueryEntryData is used as the return value of the database gateway. It is the consequence of a query.

In this case, the controller accepts the user's inputs and there's no query happens.

Instead, you should pass the information of the newly created entry: date, value, currency, description, from, to, and location.

Also, `billId` should not be a parameter. To get the current bill id, one can use `User.getInstance().getCurrentBillId()`.